core/modules/comment/comment.theme.inc

```diff
@@ -145,5 +148,8 @@ function template_preprocess_comment(&$variables) {
       $variables['classes'][] = 'comment-by-viewer';
     }
   }
+  if ($variables['title_options'] == COMMENT_TITLE_HIDDEN) {
+    $variables['classes'][] = 'hide-comment-title';
```

Comment:
What I'd find cool here for easier styling: be more specific with the css class:

- comment-title-generated
- comment-title-custom
- comment-title-hidden

I belief, that would provide maximum flexibility for themers.

Reply:
@indigoxela - I'm not seeing the use case for a themer to care the difference between generated and custom titles. In my eyes, the only thing that matters is hidden or not. However, I have no objection to providing the additional classes.

I'll put them in here and see what folks think.

Reply:
> I'm not seeing the use case for a themer to care the difference between generated and custom titles.

It doesn't hurt to provide css classes like that, simply for more flexibility for themers. If the css classes are there, they can decide how a generated or a human-provided title might look like without the need for an additional hook in their template.php (which might be a bit of a hurdle for front-enders with low or no experience with php).

Reply:
Understood. I did add three classes you suggested.